cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -81,8 +82,8 @@
   private final Map<String, LocalizeResource> resourcesToLocalize;
   private final Supplier<MapReduceTaskContextProvider> taskContextProviderSupplier;
 
-  private InputFormatProvider inputFormatProvider;
-
+  private Map<String, InputFormatProvider> inputFormatProviders;
+  private HashMap<String, Class<? extends Mapper>> mapperClasses;
```

Comment:
Type should be Map. In fact, I think it's better to have one `Map<String, MapperInput>`, which `MapperInput` contains both the `InputFormatProvider` and the mapper `Class` inside.